test/puppetlabs/puppetdb/http/command_test.clj

```diff
@@ -139,3 +137,111 @@
         (let [timestamp (get-in good-msg [:headers :received])]
           (time/parse (time/formatters :date-time) timestamp))))))
 
+(deftest wrap-with-request-normalization-all-params
+  (let [normalize (#'tgt/wrap-with-request-normalization identity)]
+    ;; Make sure that when all three params are present, the body is passed on.
+    (let [before-params {"certname" "x" "command" "y_z" "version" "1"
+                         "received" (kitchensink/timestamp)}
+          before {:params before-params
+                  :body (ByteArrayInputStream. (byte-array 0))}
+          after (normalize before)
+          after-params (:params after)]
+      ;; Accomodate the fact that command and version will be transformed.
+      (is (identical? (:body before) (:body after)))
+      (is (= (dissoc before :params) (dissoc after :params)))
```

Comment:
The identical? is testing that the function passes the *exact* pointer straight through, i.e. we hopefully don't do any work on it (but we're just indirectly checking that by seeing that it's passed through - so not a definitive test).  the latter is checking that excepting params, before and after (as a whole) are the same, wrt "=", though not identical?.  i.e. no new map entries are inserted, etc.